uni/lib/view/home/widgets/bus_stop_card.dart

```diff
@@ -44,132 +98,64 @@ class BusStopCard extends GenericCard {
+  List<Widget> getEachBusStopInfo(
+    BuildContext context,
+    Map<String, BusStopData> stopData,
+  ) {
+    return stopData.entries
+        .where(
+          (busStop) =>
+              busStop.value.trips.isNotEmpty && busStop.value.favorited,
+        )
+        .map(
+          (busStop) => Container(
+            padding: const EdgeInsets.only(top: 12),
+            child: BusStopRow(
+              stopCode: busStop.key,
+              trips: busStop.value.trips,
+              singleTrip: true,
             ),
           ),
-        ],
+        )
+        .toList()
+      ..add(
+        <Widget>[const LastUpdateTimeStamp<BusStopProvider>()] as Container,
       );
   }
-}
-
-/// Returns a widget for the title of the bus stops card
-Widget getCardTitle(BuildContext context) {
-  return Row(
-    children: <Widget>[
-      const Icon(Icons.directions_bus),
-      Text(
-        S.of(context).stcp_stops,
-        style: Theme.of(context).textTheme.titleMedium,
-      ),
-    ],
-  );
-}
-
-/// Returns a widget for all the bus stops info
-Widget getBusStopsInfo(
-  BuildContext context,
-  Map<String, BusStopData> stopData,
-) {
-  if (stopData.isNotEmpty) {
-    return Container(
-      padding: const EdgeInsets.all(4),
-      child: Column(
-        children: getEachBusStopInfo(context, stopData),
-      ),
-    );
-  } else {
-    return Center(
-      child: Text(
-        S.of(context).no_data,
-        maxLines: 2,
-        overflow: TextOverflow.fade,
-      ),
-    );
-  }
-}
-
-/// Returns a list of widgets for each bus stop info that exists
-List<Widget> getEachBusStopInfo(
-  BuildContext context,
-  Map<String, BusStopData> stopData,
-) {
-  final rows = <Widget>[const LastUpdateTimeStamp<BusStopProvider>()];
 
-  stopData.forEach((stopCode, stopInfo) {
-    if (stopInfo.trips.isNotEmpty && stopInfo.favorited) {
-      rows.add(
+  /// Returns the bus stop info if it has trips
+  Widget getCardContent(
+    BuildContext context,
+    Map<String, BusStopData> stopData,
+  ) {
+    return Column(
+      children: <Widget>[
+        getCardTitle(context),
         Container(
-          padding: const EdgeInsets.only(top: 12),
-          child: BusStopRow(
-            stopCode: stopCode,
-            trips: stopInfo.trips,
-            singleTrip: true,
+          padding: const EdgeInsets.all(4),
+          child: Column(
+            children: getEachBusStopInfo(
+              context,
+              stopData,
+            ),
           ),
         ),
-      );
-    }
-  });
+      ],
+    );
+  }
 
-  return rows;
+  /// Returns a widget for the title of the bus stops card
+  Widget getCardTitle(BuildContext context) {
+    return Row(
+      children: <Widget>[
+        const Icon(Icons.directions_bus), // color lightgrey
+        Text(
+          'STCP - Próximas Viagens',
+          style: Theme.of(context).textTheme.titleMedium,
+        ),
+      ],
+    );
+  }
```

Comment:
You can remove this method and inline it. Also, consider removing the color comment (which is not true since it depends on the global theme) and make sure to translate the string